### PR TITLE
feat: initial version

### PR DESCRIPTION
**Pull request by tmelliottjr**
- [x] Implement features. Create separate `feat: ...` commits for each feature of the initial version
- [x] 100% test coverage
- [x] Create npm token at `https://www.npmjs.com/settings/<your npm username>/tokens/create` (with "Read and Publish" selected) and add it as `NPM_TOKEN` at Then create secret at https://github.com/tmelliottjr/octokit-plugin-unique-issue/settings/secrets

### CODE REVIEW

**Review by mrsyskdr — APPROVED** [review]
Git

**Comment by tmelliottjr** [comment]
@gr2m Care taking a peek at this when you have the time? 🤗

File: .github/workflows/update-prettier.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "lts/*"
+          cache: npm
+      - run: npm ci
+      - run: npm run lint:fix
+      - uses: gr2m/create-or-update-pull-request-action@v1.x
+        env:
+          GITHUB_TOKEN: ${{ secrets.OCTOKITBOT_PAT }}
```
Comment (gr2m) [inline .github/workflows/update-prettier.yml:20]: Hm I think this should have been 

```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

when the repository is created outside of @octokit. Probably a bug in https://github.com/octokit/create-octokit-project.js/

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { Octokit } from "@octokit/core";
+import { VERSION } from "./version";
+import { Endpoints } from "@octokit/types";
+
+const OCTOKIT_UNIQUE_ISSUE_ID_PREFIX = "octokit-unique-issue id=";
+
+type CreateOrUpdateUniqueIssueOptionsT =
+  Endpoints["POST /repos/{owner}/{repo}/issues"]["parameters"] & {
+    identifier: string;
+    update_previous?: boolean;
```
Comment (gr2m) [inline src/index.ts:10]: The README still says `close_previous`, while the code implements `update_previous`. Intuitively I think `close_previous` is better, I have to enable something in order for the plugin to do something. Setting `update_previous: false` in order for the plugin to close the previous issue doesn't feel quite right.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+      if (total_count === 1 && update_previous) {
+        return await octokit.request(
+          "PATCH /repos/{owner}/{repo}/issues/{issue_number}",
+          {
+            owner,
+            repo,
+            issue_number: items[0].number,
+            ...(body !== undefined
+              ? { body: body + `\n${commentMarker}` }
```
Comment (gr2m) [inline src/index.ts:56]: would it make sense to add two line breaks to have an empty line above the marker?

```suggestion
              ? { body: body + `\n\n${commentMarker}` }
```

**Review by gr2m — COMMENTED** [review]
it would be great to export a `composeUniqueIssue` method to follow the pattern of other @octokit plugin, but that can be addressed in a follow up PR

File: src/index.ts (hunk 2 above)
Comment (tmelliottjr) [reply src/index.ts:10]: Yeah, I agree. I'll update.

File: src/index.ts (hunk 3 above)
Comment (tmelliottjr) [reply src/index.ts:56]: We talked about this and I think this makes sense! 👍🏻

**Review by gr2m — APPROVED** [review]
let's go!

**Comment by gr2m** [comment]
@tmelliottjr looks like the token you created is not an automation token? The CI requests an OTP: https://github.com/tmelliottjr/octokit-plugin-unique-issue/actions/runs/3550702560/jobs/5964322485#step:6:92

Also likes like the URL added to the initial PR is no longer valid, should be

```diff
- https://www.npmjs.com/settings/<your npm username>/tokens/create
- https://www.npmjs.com/settings/<your npm username>/tokens/new
```

The PR should be cleare that the "Automation" type must be selected

image

Would you like to send another PR to https://github.com/octokit/create-octokit-project.js/ to improve that?

To fix the release, create a new token with type=automation, update the secret in the repository, then just re-run the failed build

**Comment by tmelliottjr** [comment]
Already done! I'll open a PR.

**Comment by gr2m** [comment]
oh it looks like the build did create the github release. You'll have to delete the release and the git tag before re-running the build, otherwise it will end up with a conflict or be a noop

**Pull request merged**